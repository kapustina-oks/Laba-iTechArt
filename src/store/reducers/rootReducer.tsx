import { combineReducers } from "redux";
import authReducer from "@/store/reducers/authReducer";
import modalReducer from "@/store/reducers/modalReducer";
import userNameReducer from "@/store/reducers/userNameReducer";
import adminReducer from "@/store/reducers/adminReducer";
import editModalReducer from "@/store/reducers/editModalReducer";
import fetchReducer from "@/store/reducers/fetchReducer";
import cartReducer from "@/store/reducers/cartReducer";

const rootReducer = combineReducers({
  auth: authReducer,
  modal: modalReducer,
  userName: userNameReducer,
  admin: adminReducer,
  editModal: editModalReducer,
  reqGame: fetchReducer,
  cart: cartReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export default rootReducer;
