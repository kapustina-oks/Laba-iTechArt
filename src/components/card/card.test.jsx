import Card from "./card";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "../../store/reducers/rootReducer";

import dataGames from "../../mock/dataBase";

describe("Card", () => {
  const store = configureStore({ reducer: rootReducer });

  it("renders without crashing", () => {
    render(<Provider store={store}><Card game={dataGames[0]}/></Provider>);
    expect(screen.getByText("Game of Thrones")).toBeInTheDocument();
  });

  it("card snapshot", () => {
    render(<Provider store={store}><Card game={dataGames[0]} /></Provider>);
    expect(document.body).toMatchSnapshot();
  });

  it("", () => {
    render(<Provider store={store}><Card game={dataGames[0]} /></Provider>);
    expect(screen.queryByText("hbhh")).toBeNull();
  });
});
