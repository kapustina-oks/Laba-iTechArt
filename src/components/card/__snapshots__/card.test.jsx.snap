// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card card snapshot 1`] = `
<body>
  <div>
    <div
      class="scene"
    >
      <div
        class="card"
      >
        <div
          class="card-front"
        >
          <div
            class="product-img"
          >
            <img
              alt="game_img"
              class="size_img"
              src="https://s2.gaming-cdn.com/images/products/5848/orig/game-steam-a-game-of-thrones-genesis-cover.jpg"
            />
          </div>
          <div
            class="card-padding"
          >
            <div
              class="d_flex"
            >
              <span>
                Game of Thrones
              </span>
              <span>
                50$
              </span>
            </div>
            <div>
              <i
                class="material-icons"
              >
                 star_border star_border star_border star_border
              </i>
            </div>
          </div>
        </div>
        <div
          class="card-back text"
        >
          "A Game of Thrones - Genesis" immerses you into the heart of the battles and intrigues between the Houses that shaped the Kingdom of Westeros. From Nymeria's arrival in the Kingdom of Dorne to the awakening of the "Others" beyond the Wall, you'll live the origins of A Game of Thrones saga through more than 1000 years of history, by taking part in Westeros' founding events and largest battles. 
          <div
            class="btn-group-card"
          >
            <button
              class="btn1"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
